cypress/integration/regression/main-table-regression.spec.ts

```diff
@@ -0,0 +1,226 @@
+    //number of variants changes
+    mainTablePage.mainTable.numVariants.haveText('Variants: 2')
+    mainTablePage.mainTable.addGene.click()
+    mainTablePage.mainTable.searchFilter.type('CHSY1')
+    mainTablePage.mainTable.geneCheckbox.check()
+    mainTablePage.mainTable.applyButton.click()
+    mainTablePage.mainTable.numVariants.haveText('Variants: 1')
+    mainTablePage.mainTable.tableRow.getButtonByText('p.Y434=').click()
+    variantDrawerPage.variantDrawer.addTag.eq(1).click()
+    variantDrawerPage.variantDrawer.tagInput.type('Custom_tag')
+    variantDrawerPage.variantDrawer.addCustomTag.forceClick()
+    cy.wait(500)
+    cy.intercept('POST', '/app/ws_tags').as('addTags')
+    variantDrawerPage.variantDrawer.saveTags.forceClick()
+    cy.wait('@addTags', { timeout: Timeouts.TwentySecondsTimeout })
+  })
+
+  it('should add note to the variant | step 5', () => {
+    mainTablePage.visit(link)
+    cy.intercept('POST', '/app/tab_report').as('dsUpload')
+    cy.wait('@dsUpload', { timeout: Timeouts.TwentySecondsTimeout })
+    mainTablePage.mainTable.selectPreset.click()
+    cy.intercept('POST', '/app/tab_report').as('loadPreset')
+    mainTablePage.mainTable.numVariants.haveText('Variants: 2592')
+    mainTablePage.mainTable.preset.getButtonByText(homozygous).click()
+    cy.wait('@loadPreset', { timeout: Timeouts.TwentySecondsTimeout })
+    //number of variants changes
+    mainTablePage.mainTable.numVariants.haveText('Variants: 2')
+    mainTablePage.mainTable.addGene.click()
+    mainTablePage.mainTable.searchFilter.type('CHSY1')
+    mainTablePage.mainTable.geneCheckbox.check()
+    mainTablePage.mainTable.applyButton.click()
+    mainTablePage.mainTable.numVariants.haveText('Variants: 1')
+    mainTablePage.mainTable.tableRow.getButtonByText('p.Y434=').click()
+    variantDrawerPage.variantDrawer.addNote.click()
+    variantDrawerPage.variantDrawer.fillSpace.type('note addition test')
+    cy.intercept('POST', '/app/ws_tags?ds=PGP3140_wgs_panel_hl&rec=**').as(
+      'addNote',
+    )
+    variantDrawerPage.variantDrawer.saveNote.click()
+    cy.wait('@addNote', { timeout: Timeouts.TwentySecondsTimeout })
+  })
+
+  it('should filter main table by custom tag | step 6', () => {
+    mainTablePage.visit('/ws?ds=PGP3140_wgs_panel_hl')
+    cy.intercept('POST', '/app/tab_report').as('dsUpload')
+    cy.wait('@dsUpload', { timeout: Timeouts.TwentySecondsTimeout })
+    mainTablePage.mainTable.selectPreset.click()
+    cy.intercept('POST', '/app/tab_report').as('loadPreset')
+    mainTablePage.mainTable.numVariants.haveText('Variants: 2592')
+    mainTablePage.mainTable.preset.getButtonByText(homozygous).click()
+    cy.wait('@loadPreset', { timeout: Timeouts.TwentySecondsTimeout })
+    //number of variants changes
+    mainTablePage.mainTable.numVariants.haveText('Variants: 2')
+    mainTablePage.mainTable.addTag.first().click()
+    mainTablePage.mainTable.searchFilter.type('CustomTag')
```

Comment:
`TestData` here too. You can generate test data in the beginning and reuse during the test.